tests/integration/cmd_describe_component_test.go

```diff
@@ -597,4 +597,93 @@ var _ = Describe("odo describe component command tests", func() {
 			})
 		}
 	})
+	Context("checking for remote source code location", func() {
+		for _, podman := range []bool{true, false} {
+			podman := podman
+			for _, ctx := range []struct {
+				devfile, title string
+				checker        func(output string, isJSON bool)
+				beforeEach     func()
+			}{
+				{
+					title:   "devfile with sourceMapping",
+					devfile: "devfileSourceMapping.yaml",
+					checker: func(output string, isJSON bool) {
+						const location = "/test"
+						if isJSON {
+							helper.JsonPathContentIs(output, "devfileData.devfile.components.#(name==runtime).container.sourceMapping", location)
+							return
+						}
+						Expect(output).To(ContainSubstring(location))
+					},
+				},
+				{
+					devfile: "devfile.yaml",
+					title:   "devfile with no sourceMapping, defaults to /projects",
+					checker: func(output string, isJSON bool) {
+						const location = "/projects"
+						if isJSON {
+							helper.JsonPathContentIs(output, "devfileData.devfile.components.#(name==runtime).container.sourceMapping", location)
+							return
+						}
+						Expect(output).To(ContainSubstring(location))
+					},
+				},
+				{
+					devfile: "devfileCompositeBuildRunDebugInMultiContainersAndSharedVolume.yaml",
+					title:   "devfile with containers that has mountSource set to false",
+					checker: func(output string, isJSON bool) {
+						if isJSON {
+							helper.JsonPathContentIs(output, "devfileData.devfile.components.#(name==runtime).container.sourceMapping", "/projects")
+							helper.JsonPathDoesNotExist(output, "devfileData.devfile.components.#(name==sleeper-run).container.sourceMapping")
+							helper.JsonPathDoesNotExist(output, "devfileData.devfile.components.#(name==sleeper-build).container.sourceMapping")
+							helper.JsonPathDoesNotExist(output, "devfileData.devfile.components.#(name==echo-er).container.sourceMapping")
+							helper.JsonPathDoesNotExist(output, "devfileData.devfile.components.#(name==build-checker).container.sourceMapping")
+							return
+						}
+						Expect(output).To(ContainSubstring("runtime\n    Source Mapping: /projects"))
+						helper.DontMatchAllInOutput(output, []string{"sleeper-run\n    Source Mapping: /projects", "sleeper-build\n    Source Mapping:", "echo-er\n    Source Mapping:", "build-checker\n    Source Mapping:"})
+					},
+				},
+			} {
+				ctx := ctx
+				When(fmt.Sprintf("using %s and starting an odo dev session", ctx.title), helper.LabelPodmanIf(podman, func() {
+					var devSession helper.DevSession
+					BeforeEach(func() {
+						helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+						helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", ctx.devfile)).ShouldPass()
```

Comment:
One last thing: can you use the random `cmpName` as the component name instead of the hardcoded `aname`? This will help when running Podman tests in parallel (https://github.com/redhat-developer/odo/pull/6499).